### PR TITLE
Use $0 as ending position in code snippets

### PR DESCRIPTION
**Pull request by doggy8088**
#2605

### CODE REVIEW

**Review by iankingh — COMMENTED** [review]
LGTM

**Review by jdneo — APPROVED** [review]
LGTM. Now when using `tab` to switch the focus, we won't 'delete' code unexpectedly.

**Review by iankingh — APPROVED** [review]
LGTM

File: snippets/java.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
 		"prefix": "ctor",
 		"body": [
 			"public ${1:${TM_FILENAME_BASE}}($2) {",
-			"\t${0:super();}",
+			"\t${3:super();}$0",
```
Comment (rgrunber) [inline snippets/java.json:15]: Agreed here. The old behaviour basically overwrites the `super()` invocation.

**Pull request merged**